tsl/test/sql/continuous_aggs_multi.sql

```diff
@@ -98,158 +96,58 @@ select create_hypertable('continuous_agg_test', 'timeval', chunk_time_interval=>
 CREATE OR REPLACE FUNCTION integer_now_test1() returns int LANGUAGE SQL STABLE as $$ SELECT coalesce(max(timeval), 0) FROM continuous_agg_test $$;
 SELECT set_integer_now_func('continuous_agg_test', 'integer_now_test1');
 
-
 INSERT INTO continuous_agg_test VALUES
-    (10, - 4, 1), (11, - 3, 5), (12, - 3, 7), (13, - 3, 9), (14,-4, 11),
+    (10, -4, 1), (11, -3, 5), (12, -3, 7), (13, -3, 9), (14, -4, 11),
     (15, -4, 22), (16, -4, 23);
 
-CREATE MATERIALIZED VIEW cagg_1( timed, cnt ) 
-WITH ( timescaledb.continuous , timescaledb.refresh_lag = '-2', timescaledb.materialized_only = true)
+CREATE MATERIALIZED VIEW cagg_1( timed, cnt )
+WITH (timescaledb.continuous,
+      timescaledb.materialized_only = true)
 AS
-    SELECT time_bucket( 2, timeval), COUNT(col1) 
+    SELECT time_bucket( 2, timeval), COUNT(col1)
     FROM continuous_agg_test
     GROUP BY 1;
-CREATE MATERIALIZED VIEW cagg_2( timed, maxval) 
-WITH ( timescaledb.continuous , timescaledb.refresh_lag = '2', timescaledb.materialized_only = true)
+CREATE MATERIALIZED VIEW cagg_2( timed, maxval)
+WITH (timescaledb.continuous,
+      timescaledb.materialized_only = true)
 AS
-    SELECT time_bucket(2, timeval), max(col2) 
+    SELECT time_bucket(2, timeval), max(col2)
     FROM continuous_agg_test
     GROUP BY 1;
-refresh materialized view cagg_1;
+
+CALL refresh_continuous_aggregate('cagg_1', NULL, NULL);
 select * from cagg_1 order by 1;
-refresh materialized view cagg_2;
+CALL refresh_continuous_aggregate('cagg_2', NULL, 14);
 select * from cagg_2 order by 1;
 --this insert will be processed only by cagg_1 and copied over to cagg_2
-insert into continuous_agg_test values( 14, -2, 100); 
-refresh materialized view cagg_1;
+insert into continuous_agg_test values( 14, -2, 100);
+CALL refresh_continuous_aggregate('cagg_1', NULL, NULL);
 select * from cagg_1 order by 1;
-refresh materialized view cagg_2;
+CALL refresh_continuous_aggregate('cagg_2', NULL, 14);
 select * from cagg_2 order by 1;
 SET ROLE :ROLE_SUPERUSER;
 select * from _timescaledb_catalog.continuous_aggs_invalidation_threshold order by 1;
 select * from _timescaledb_catalog.continuous_aggs_materialization_invalidation_log order by 1;
 SET ROLE :ROLE_DEFAULT_PERM_USER;
 --this insert will be processed only by cagg_1 and cagg_2 will process the previous
 --one
-insert into continuous_agg_test values( 18, -2, 200); 
-refresh materialized view cagg_1;
+insert into continuous_agg_test values( 18, -2, 200);
+CALL refresh_continuous_aggregate('cagg_1', NULL, NULL);
 select * from cagg_1 order by 1;
-refresh materialized view cagg_2;
+CALL refresh_continuous_aggregate('cagg_2', NULL, 16);
```

Comment:
Why just to 16?

Reply:
You need to look at the output log to see that this is where the previous materializer stopped.

Reply:
Could you add a comment?